eth_defi/aave_v3/events.py

```diff
@@ -0,0 +1,311 @@
+                "variable_borrow_rate",
+                "liquidity_index",
+                "variable_borrow_index",
+            ],
+            "decode_function": decode_reserve_data_updated,
+        },
+    }
+
+
+def _decode_base(log: LogResult) -> dict:
+    block_time = datetime.datetime.utcfromtimestamp(log["timestamp"])
+
+    return {
+        "block_number": int(log["blockNumber"], 16),
+        "timestamp": block_time.isoformat(),
+        "tx_hash": log["transactionHash"],
+        "log_index": int(log["logIndex"], 16),
+    }
+
+
+def decode_reserve_data_updated(aave_network_name: str, log: LogResult) -> dict:
+    """Process a ReserveDataUpdated event. The event signature is:
+
+    .. code-block::
+
+        # topic0: signature 0x804c9b842b2748a22bb64b345453a3de7ca54a6ca45ce00d415894979e22897a
+        event ReserveDataUpdated(
+            address indexed reserve,    # topic1
+            uint256 liquidityRate,      # data0
+            uint256 stableBorrowRate,   # data1
+            uint256 variableBorrowRate, # data2
+            uint256 liquidityIndex,     # data3
+            uint256 variableBorrowIndex # data4
+        );
+    """
+    # Ensure the event comes from the correct smart contract
+    if log["address"].lower() != AAVE_V3_NETWORKS[aave_network_name.lower()].pool_address.lower():
+        return None
+
+    # Do additional lookup for the token data
+    # web3 = log["event"].web3
+    # token_cache: TokenCache = log["context"]
+    result = _decode_base(log)
+
+    # Any indexed Solidity event parameter will be in topics data.
+    # The first topics (0) is always the event signature.
+    if len(log["topics"]) < 2:
+        logging.warn(f'IGNORING EVENT: block={log["blockNumber"]} tx={log["transactionHash"]} topics={log["topics"]} data={log["data"]}')
+        return None
+    event_signature, reserve = log["topics"]
+    deposit_address = convert_uint256_string_to_address(reserve)
+
+    # Any non-indexed Solidity event parameter will be in the data section.
+    # Chop data blob to byte32 entries
+    liquidity_rate_raw, stable_borrow_rate_raw, variable_borrow_rate_raw, liquidity_index_raw, variable_borrow_index_raw = decode_data(log["data"])
+
+    liquidity_rate = convert_int256_bytes_to_int(liquidity_rate_raw)
+    stable_borrow_rate = convert_int256_bytes_to_int(stable_borrow_rate_raw)
+    variable_borrow_rate = convert_int256_bytes_to_int(variable_borrow_rate_raw)
+    liquidity_index = convert_int256_bytes_to_int(liquidity_index_raw)
+    variable_borrow_index = convert_int256_bytes_to_int(variable_borrow_index_raw)
+
+    result.update({
+        "reserve": deposit_address,
+        "liquidity_rate": liquidity_rate,
+        "stable_borrow_rate": stable_borrow_rate,
+        "variable_borrow_rate": variable_borrow_rate,
+        "liquidity_index": liquidity_index,
+        "variable_borrow_index": variable_borrow_index,
+        "contract_address": log["address"],
+    })
```

Comment:
(nit)
No need to create and intermediary dict just to pass it to `update()` and then unpack it, we could send the updated values as kwargs directly.

It's a very minor thing, though, so please don't bother if you don't feel like. :)